linking problems with CListModeDataGEHDF5

#1188 move the implementations of `GEHDF5ListmodeInputFileFormat` from `.h` to `.cxx`. Unfortunately this has increased "recursiveness" between IO and listmode libraries (i.e. they both need eachother). This leads to linking errors for SIRF, see e.g.
https://github.com/SyneRBI/SIRF/actions/runs/4866834387/jobs/8841295908
```
/usr/bin/ld: /home/runner/install/lib/libIO.a(GEHDF5ListmodeInputFileFormat.cxx.o): in function `stir::GE::RDF_HDF5::GEHDF5ListmodeInputFileFormat::read_from_file(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
GEHDF5ListmodeInputFileFormat.cxx:(.text+0x83): undefined reference to `stir::GE::RDF_HDF5::CListModeDataGEHDF5::CListModeDataGEHDF5(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
```

--- a/src/IO/CMakeLists.txt
+++ b/src/IO/CMakeLists.txt
@@ -70,7 +70,7 @@ endif()
 if (HAVE_HDF5)
  list(APPEND ${dir_LIB_SOURCES}
     GEHDF5Wrapper.cxx
-    GEHDF5ListmodeInputFileFormat
+    GEHDF5ListmodeInputFileFormat.cxx
  )
 endif()
 
@@ -132,4 +132,4 @@ endif()
 
 # currently needed for ParametricDensity (TODO get rid of this somehow?)
 target_link_libraries(IO modelling_buildblock )
-
+target_link_libraries(IO listmode_buildblock)

--- a/src/cmake/FindJANSSON.cmake
+++ b/src/cmake/FindJANSSON.cmake
@@ -54,7 +54,7 @@ elseif (JANSSON_INCLUDE_DIR AND EXISTS "${JANSSON_INCLUDE_DIR}/jansson.h")
 endif ()
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(Jansson
+find_package_handle_standard_args(JANSSON
         REQUIRED_VARS JANSSON_LIBRARY JANSSON_INCLUDE_DIR
         VERSION_VAR JANSSON_VERSION_STRING)
 

--- a/src/cmake/stir_dirs.cmake
+++ b/src/cmake/stir_dirs.cmake
@@ -40,15 +40,13 @@ ${PROJECT_SOURCE_DIR}/src/spatial_transformation_buildblock/spatial_transformati
 ${PROJECT_SOURCE_DIR}/src/scatter_buildblock/scatter_registries.cxx
 )
 
-SET( STIR_LIBRARIES analytic_FBP3DRP analytic_FBP2D       iterative_OSMAPOSL   iterative_KOSMAPOSL
+# need to list IO first such that its dependencies on other libraries are resolved
+SET( STIR_LIBRARIES IO analytic_FBP3DRP analytic_FBP2D       iterative_OSMAPOSL   iterative_KOSMAPOSL
      iterative_OSSPS
       scatter_buildblock modelling_buildblock listmode_buildblock recon_buildblock  
-      display  IO  data_buildblock numerics_buildblock  buildblock 
+      display   data_buildblock numerics_buildblock buildblock
       spatial_transformation_buildblock
       Shape_buildblock eval_buildblock 
-      # repeat for linking
-      numerics_buildblock modelling_buildblock listmode_buildblock
-      IO modelling_buildblock IO buildblock
 )
 
 #copy to PARENT_SCOPE

--- a/src/listmode_buildblock/CMakeLists.txt
+++ b/src/listmode_buildblock/CMakeLists.txt
@@ -57,4 +57,4 @@ endif()
 
 include(stir_lib_target)
 
-target_link_libraries(listmode_buildblock IO data_buildblock )
+target_link_libraries(listmode_buildblock data_buildblock )

